server/src/recipe/middleware/recipe.authorization.service.ts

```diff
@@ -0,0 +1,76 @@
+import { NextFunction, Request, Response } from "express";
+import {
+  ApiError, ERRORS, SocoboRequest
+} from "../../app/index";
+
+import { DbExtension } from "../../db/interface/db-extension";
+import { SocoboUser } from "../../socobouser/index";
+import { Recipe } from "../index";
+
+export class RecipeAuthorizationService {
```

Comment:
what think you about to make this as a middleware? it's not really needed, but it would looks better and you doesn't need the SocoboUserRequest inside your RecipeHandler!? :-)

Reply:
I will check if its possible in all cases

Reply:
I'm not sure how I should do this without repeating a lot of code from the handler methods. Because I need to fetch the recipe before I can check if the user is authorized. So if I want to use a middleware I would need to fetch the recipe in the middleware and then check the authorization. After this I would need to fetch the recipe again in the handler to do the remaining work. Maybe I could add the recipe / recipes to the request, but there are case where I need different recipe data in the handler (e.g. getAllRecipes and search). Any Idea?

Reply:
Yes you could add this to the request. But your solution is good. We should use it and move on.